src/base/globals.lua

```diff
@@ -73,14 +75,30 @@
 --    global package.loaded table.
 ---
 
-	premake.override(_G, "require", function(base, modname, versions)
-		local result, mod = pcall(base,modname)
-		if not result then
-			error( mod, 3 )
+	p.override(_G, "require", function(base, modname, versions)
+		local ok, mod = pcall(base, modname)
+		if not ok then
+			error(mod, 3)
 		end
-		if mod and versions and not premake.checkVersion(mod._VERSION, versions) then
+
+		if mod and versions and not p.checkVersion(mod._VERSION, versions) then
 			error(string.format("module %s %s does not meet version criteria %s",
 				modname, mod._VERSION or "(none)", versions), 3)
 		end
+
 		return mod
 	end)
+
+
+---
+-- Returns the specified module if it exists, or `nil` otherwise.
+---
+
+	function softrequire(modname, versions)
```

Comment:
This will also return 'nil' if the version is incorrect... it will however still have loaded the module.